server/console_api.py

```
@@ -143,7 +144,17 @@ async def add_group(request: Request, user=Depends(admin_only)):
                 "INSERT INTO groups (id, label, role, sort) VALUES (?, ?, ?, ?)",
                 (gid, label, role, (last or 0) + 1),
             )
-        except Exception:
+        except sqlite3.IntegrityError:
+            # ‏rollback לפני ה-raise, ולא אחרי (#184). ה-INSERT שנכשל
+            # כבר פתח טרנזאקציית כתיבה; חריגה שיוצאת בלי לסגור אותה
```

Ruff (0.16.2)

[error] Contains control characters that can permit obfuscated code

(PLE2502)